help/en/html/doc/Technical/NetBeans.shtml

```diff
@@ -28,713 +28,701 @@
-              in the <em>Checkout Selected Revision</em> box, and click <b class="button">Checkout</b>
-              </td>
-              </tr>
-              </table>
-              </div>
-              </div>
-              </div>
-              </div>
-              <div id="footer">
-              <div id="footer-text">
-              Last updated 2020-02-29 14:36:14 -0500
-              </div>
-              </div>
-                <!--#include virtual="/Footer.shtml" -->
+            <!-- this was created by running the command `asciidoctorj -a toc NetBeans.adoc`
+                and copying the contents of body element in the resultant NetBeans.html file -->
+            <div id="header">
+            <h1>JMRI Code: Building with Apache NetBeans</h1>
+            <div id="toc" class="toc">
+            <div id="toctitle">Table of Contents</div>
+            <ul class="sectlevel1">
+            <li><a href="#_introduction">Introduction</a></li>
+            <li><a href="#_installation">Installation</a>
+            <ul class="sectlevel2">
+            <li><a href="#_linux">Linux</a></li>
+            <li><a href="#_macos">macOS</a></li>
+            <li><a href="#_windows">Windows</a></li>
+            </ul>
+            </li>
+            <li><a href="#_using_apache_netbeans_with_jmris_git_code_repository">Using Apache NetBeans with JMRI&#8217;s Git Code Repository</a>
+            <ul class="sectlevel2">
+            <li><a href="#_checking_out_code">Checking out code</a></li>
+            <li><a href="#_local_commits">Local Commits</a></li>
+            <li><a href="#_updating_the_code_from_git">Updating the code from Git</a></li>
+            <li><a href="#_creating_a_pull_request">Creating a Pull Request</a></li>
+            </ul>
+            </li>
+            <li><a href="#_building_and_running_jmri">Building and running JMRI</a>
+            <ul class="sectlevel2">
+            <li><a href="#_prime">Prime</a></li>
+            <li><a href="#_build">Build</a></li>
+            <li><a href="#_run_an_application">Run an Application</a></li>
+            <li><a href="#_debug_an_application">Debug an Application</a></li>
+            <li><a href="#_profile_an_application">Profile an Application</a></li>
+            </ul>
+            </li>
+            <li><a href="#_making_a_jar_file">Making a Jar File</a></li>
+            <li><a href="#_making_a_release_package_for_windows_using_nsis">Making a release package for Windows using NSIS</a></li>
+            <li><a href="#_using_apache_netbeans_xml_tools">Using Apache NetBeans XML tools</a></li>
+            <li><a href="#_running_spotbugs">Running SpotBugs</a></li>
+            <li><a href="#_loading_a_specific_branch_from_github">Loading a Specific Branch from GitHub</a></li>
+            </ul>
+            </div>
+            </div>
+            <div id="content">
+            <div class="sect1">
+            <h2 id="_introduction">Introduction</h2>
```

Comment:
The targets in this file have` id` attributes, but don't have the `name` attributes[ needed by JavaHelp](http://jmri.org/help/en/html/doc/Technical/Help.shtml)

Reply:
I put the targets in and CI complained of invalid HTML. Since some notes on how to use the NetBeans IDE will not help a JMRI user deal with JMRI, I don't think this page need be concerned with working in JavaHelp.

Reply:
`gsed -i -e 's/<\(.*\) id=\"\(_.*\)\">/\<\1>\<a id=\"\2\" name=\"\2\">\<\/a>/' NetBeans.shtml` (plain sed on linux) will find all elements with just an id attribute starting with an underscore (_) and create an anchor element for it. (Like is created by `asciidoctorj -a toc NetBeans.adoc`.)